tests/test_pdf.py

```diff
@@ -182,6 +202,39 @@ def test_pdf_integration_staterror(backend):
     )
 
 
+def test_poiless_model(backend):
+    spec = {
+        'channels': [
+            {
+                'name': 'channel',
+                'samples': [
+                    {
+                        'name': 'goodsample',
+                        'data': [10.0],
+                        'modifiers': [
+                            {
+                                'type': 'normsys',
+                                'name': 'shape',
+                                'data': {"hi": 0.5, "lo": 1.5},
+                            }
+                        ],
+                    },
+                ],
+            }
+        ]
+    }
+    m = pyhf.Model(spec, poi_name=None)
+
+    data = [12] + m.config.auxdata
+    pyhf.infer.mle.fit(data, m)
+
+    with pytest.raises(RuntimeError):
+        pyhf.infer.mle.fixed_poi_fit(1.0, data, m)
+
+    with pytest.raises(RuntimeError):
+        pyhf.infer.hypotest(1.0, data, m)
```

Comment:
```suggestion
    model = pyhf.Model(spec, poi_name=None)

    data = [12] + model.config.auxdata
    pyhf.infer.mle.fit(data, model)

    with pytest.raises(RuntimeError):
        pyhf.infer.mle.fixed_poi_fit(1.0, data, model)

    with pytest.raises(RuntimeError):
        pyhf.infer.hypotest(1.0, data, model)
```

Nitpicky nitpick of preferring variable names being longer than 1 letter.